src/Components/Aspire.Confluent.Kafka/PublicAPI.Unshipped.txt

```diff
@@ -19,7 +19,27 @@ Aspire.Confluent.Kafka.KafkaProducerSettings.DisableMetrics.get -> bool
 Aspire.Confluent.Kafka.KafkaProducerSettings.DisableMetrics.set -> void
 Microsoft.Extensions.Hosting.AspireKafkaConsumerExtensions
 Microsoft.Extensions.Hosting.AspireKafkaProducerExtensions
-static Microsoft.Extensions.Hosting.AspireKafkaConsumerExtensions.AddKafkaConsumer<TKey, TValue>(this Microsoft.Extensions.Hosting.IHostApplicationBuilder! builder, string! connectionName, System.Action<Aspire.Confluent.Kafka.KafkaConsumerSettings!>? configureSettings = null, System.Action<Confluent.Kafka.ConsumerBuilder<TKey, TValue>!>? configureBuilder = null) -> void
-static Microsoft.Extensions.Hosting.AspireKafkaConsumerExtensions.AddKeyedKafkaConsumer<TKey, TValue>(this Microsoft.Extensions.Hosting.IHostApplicationBuilder! builder, string! name, System.Action<Aspire.Confluent.Kafka.KafkaConsumerSettings!>? configureSettings = null, System.Action<Confluent.Kafka.ConsumerBuilder<TKey, TValue>!>? configureBuilder = null) -> void
-static Microsoft.Extensions.Hosting.AspireKafkaProducerExtensions.AddKafkaProducer<TKey, TValue>(this Microsoft.Extensions.Hosting.IHostApplicationBuilder! builder, string! connectionName, System.Action<Aspire.Confluent.Kafka.KafkaProducerSettings!>? configureSettings = null, System.Action<Confluent.Kafka.ProducerBuilder<TKey, TValue>!>? configureBuilder = null) -> void
-static Microsoft.Extensions.Hosting.AspireKafkaProducerExtensions.AddKeyedKafkaProducer<TKey, TValue>(this Microsoft.Extensions.Hosting.IHostApplicationBuilder! builder, string! name, System.Action<Aspire.Confluent.Kafka.KafkaProducerSettings!>? configureSettings = null, System.Action<Confluent.Kafka.ProducerBuilder<TKey, TValue>!>? configureBuilder = null) -> void
+static Microsoft.Extensions.Hosting.AspireKafkaConsumerExtensions.AddKafkaConsumer<TKey, TValue>(this Microsoft.Extensions.Hosting.IHostApplicationBuilder! builder, string! connectionName) -> void
```

Comment:
This is a lot of new APIs (we went from 4 to 24). Are all these needed?

Reply:
We previously have 4 api with 2 optional parameters. To ensure backwards compatibility I dropped the default values
and exposed each combination of the previous API. we went from 4 to 4 * 4 APIs.

The new overloads replace `Action<ProducerBuilder<TKey, TValue>>` and `Action<ConsumerBuilder<TKey, TValue>>` parameters by `Action<IServiceProvider, ProducerBuilder<TKey, TValue>>` and `Action<IServiceProvider, ConsumerBuilder<TKey, TValue>>` which add 4 * 2 APIs.

At both side, consumer and producer, the APIs forward all the call to the same internal implementation.